tests/legacy-cli/e2e/tests/build/auto-csp.ts

```diff
@@ -0,0 +1,131 @@
+import assert from 'node:assert';
+import { getGlobalVariable } from '../../utils/env';
+import { expectFileToMatch, rimraf, writeMultipleFiles } from '../../utils/fs';
+import { findFreePort } from '../../utils/network';
+import { installWorkspacePackages } from '../../utils/packages';
+import { execAndWaitForOutputToMatch, ng } from '../../utils/process';
+import { updateJsonFile, updateServerFileForWebpack, useSha } from '../../utils/project';
+
+const MULTI_HASH_CSP =
+  /script-src 'strict-dynamic' (?:'sha256-[^']+' )+https: 'unsafe-inline';object-src 'none';base-uri 'self';/;
+
+export default async function () {
+  assert(
+    getGlobalVariable('argv')['esbuild'],
+    'This test should not be called in the Webpack suite.',
+  );
+  // forcibly remove in case another test doesn't clean itself up
+  // await rimraf('node_modules/@angular/ssr');
+  // We don't need SSR yet but eventually for when Auto-CSP will also have a SSR implementation.
+  // await ng('add', '@angular/ssr', '--skip-confirmation', '--skip-install');
+
+  await useSha();
+  await installWorkspacePackages();
+
+  // Turn on auto-CSP
+  await updateJsonFile('angular.json', (json) => {
+    const build = json['projects']['test-project']['architect']['build'];
+    build.options = {
+      ...build.options,
+      security: { autoCsp: true },
+    };
+  });
+
+  await writeMultipleFiles({
+    'serve.js': `
+    const express = require('express');
+    const path = require('path');
+
+    const app = express();
+    const PORT = process.env.PORT || 3000;
+
+    app.use(express.static(path.join(__dirname, 'dist/test-project/browser')));
+
+    app.listen(PORT, () => {
+      console.log('Node Express server listening on ' + PORT);
+    });
+    `,
+    'src/index.html': `
+      <!doctype html>
+      <html lang="en">
+      <head>
+        <meta charset="utf-8">
+        <base href="/">
+        <script>console.log("Inline Script Head");</script>
+      </head>
+      <body>
+        <app-root></app-root>
+        <script>console.log("Inline Script Body");</script>
```

Comment:
Should we include an external script tag here too to make sure that works? You can probably put a JS file in the `src/public/` directory which will be included in the output so the browser can load it successfully.